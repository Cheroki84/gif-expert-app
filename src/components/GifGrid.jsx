import { useEffect, useState } from "react";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ gifRender }) => {
    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, [])
    const getGifs = async () => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=pokemon&limit=10&api_key=b2PjivnhvJXps3cGiDwMVRRdYvSkcGHa';
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs)
        setImages(gifs);
    }

    return (
        <div>
            <h3>{gifRender}</h3>

            {
            images.map(img => (
                <GifGridItem 
                key={img.id}
                {...img}
                />
            ))
            }

        </div>
    )
}

export default GifGrid;
