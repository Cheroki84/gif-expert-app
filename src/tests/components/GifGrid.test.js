import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas sobre el componente GifGrid', () => {
    
    const category = 'Hola Cheroki';

    test('Debe de mostrarse el componente GifGrid correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    });
    
    test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        expect(wrapper).toMatchSnapshot();

    });
    

});
