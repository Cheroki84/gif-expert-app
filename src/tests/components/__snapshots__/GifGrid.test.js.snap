// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid Debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    Hola Cheroki
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas sobre el componente GifGrid Debe de mostrarse el componente GifGrid correctamente 1`] = `
<Fragment>
  <h3>
    Hola Cheroki
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
